node/heartbeat/monitor.go

```diff
@@ -34,6 +35,7 @@ func NewMonitor(
 	marshalizer marshal.Marshalizer,
 	maxDurationPeerUnresponsive time.Duration,
 	pubKeyList []string,
+	shardID uint32,
```

Comment:
this is not used properly. Should drop the shardId parameter and field. This should be sent by the message originator (the message will have the shardId set by the one who outputs the hearbeat message)
As written now, all messages received by one node will have the same shard ID as the node on which we are connected to.

Reply:
+1

Reply:
done